actionpack/lib/action_controller/metal/http_authentication.rb

```diff
@@ -431,8 +431,9 @@ def authenticate_or_request_with_http_token(realm = "Application", message = nil
           authenticate_with_http_token(&login_procedure) || request_http_token_authentication(realm, message)
         end
 
-        # Authenticate using an HTTP Bearer token. Returns true if
-        # authentication is successful, false otherwise.
+        # Authenticate using an HTTP Bearer token.
+        # Returns the return value of <tt>login_procedure</tt> if a
+        # token is found. Returns <tt>nil</tt> if no token is found.
```

Comment:
An example could clarify things a bit here as well:
```suggestion
        # token is found. Returns <tt>nil</tt> if no token is found.
        #
        #   authenticate_with_http_token { |token, o| User.for_token(token, o) }
        #   => #<User id: 1 ..>
```

Or maybe, it's clearer to link to Token documentation as we do for `authenticate_or_request_with_http_token`.

```suggestion
        # token is found. Returns <tt>nil</tt> if no token is found.
        #
        # See ActionController::HttpAuthentication::Token for example usage.
```